core/primitives/decode_stream.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+#define FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+
+#include <string_view>
+
+#include "common/outcome.hpp"
+#include "common/hexutil.hpp"
+#include "common/textutil.hpp"
+
+namespace primitives {
+
+  /**
+   * Base class for representing data streams
+   */
+  template <typename T>
+  class DecodeStream {
+   public:
+    static constexpr auto is_decoder_stream = true; /**< Decoder type flag */
+
+    /**
+     * @brief Decode content from hex string
+     * @param data input string
+     * @return decoded data
+     */
+    DecodeStream<T> &operator>>(std::string_view data) {
+      fc::outcome::result<std::vector<uint8_t>> bytes = fc::common::unhex(data);
+      if (bytes) {
+        bytes >> *this;
+      }
+      return *this;
+    }
+
+    /**
+     * @brief Write stream content to std::string
+     * @param stream input stream
+     * @param output target string
+
+     * @return output string
+     */
+    friend std::string &operator>>(const DecodeStream<T> &stream,
+                                   std::string &output) {
+      if constexpr (std::is_integral<T>::value) {
+        output.append(textutil::convertNumberToString(stream.content_));
```

Comment:
What if it isn't integral? Is it ok to continue? Looks like error masking behavior.

Reply:
This implementation was designed to implement only numeric values, I decided to remove  converting numeric values option into strings after refactoring.